File: src/core/demo/physicsDemo.ts
import { linFanInitialCoordinate, coordinateToRecord, bigFiveFromCoordinate } from "../personality/coordinate";
import { CharacterPhysicsEngine, createCharacterPhysicsState } from "../physics/physicsEngine";
import type { ExperienceEvent } from "../event/event";

const coordinate = linFanInitialCoordinate();
const state = createCharacterPhysicsState({
  coordinate,
  personality: bigFiveFromCoordinate(coordinate),
  learningRate: 0.03
});

const event: ExperienceEvent = {
  id: "event_wangxue_no_reply",
  description: "王雪已经三天没有回复林凡的消息，今天深夜突然出现在他家门口。",
  tags: ["王雪", "失联", "等待", "亲密关系", "夜晚"],
  intensity: 0.82,
  importance: 0.86,
  relationshipWeight: 0.95,
  expectationGap: 0.78,
  personalitySensitivity: 0.9
};

const result = new CharacterPhysicsEngine().processEvent(state, event);

console.log("Character Physics TypeScript Demo");
console.log("\nInitial Coordinate:");
console.log(coordinateToRecord(coordinate));
console.log("\nImpact:");
console.log(result.impactScore);
console.log("\nEmotion:");
console.log(result.emotion);
console.log("\nImpact Particle:");
console.log({
  category: result.particle.category,
  delta: coordinateToRecord(result.particle.vector.delta),
  rationale: result.particle.vector.rationale
});
console.log("\nMemory Node:");
console.log({
  id: result.memoryNode.id,
  importance: result.memoryNode.importance,
  emotion: result.memoryNode.emotion,
  repetitionCount: result.memoryNode.repetitionCount,
  clusterId: result.memoryNode.clusterId,
  beliefEffect: result.memoryNode.beliefEffect
});
console.log("\nImpact Cluster:");
console.log({
  id: result.cluster.id,
  mass: result.cluster.mass,
  density: result.cluster.density,
  stability: result.cluster.stability,
  centerCoordinate: coordinateToRecord(result.cluster.centerCoordinate)
});
console.log("\nCoordinate Drift:");
console.log({
  before: coordinateToRecord(result.coordinateDrift.before),
  force: coordinateToRecord(result.coordinateDrift.totalForce),
  after: coordinateToRecord(result.coordinateDrift.after)
});
console.log("\nGalaxy Step Trace:");
console.log({
  totalForce: coordinateToRecord(result.galaxyStep.totalForce),
  previousVelocity: coordinateToRecord(result.galaxyStep.drift.previousVelocity),
  nextVelocity: coordinateToRecord(result.galaxyStep.drift.nextVelocity),
  clusterMetrics: result.galaxyStep.clusterMetrics.map((item) => ({
    clusterId: item.clusterId,
    category: item.category,
    mass: item.metrics.mass,
    density: item.metrics.density,
    stability: item.metrics.stability,
    variance: item.metrics.variance
  }))
});
console.log("\nBoundary Impact:");
console.log({
  incomingStress: result.boundaryImpact.incomingStress,
  overflowAmount: result.boundaryImpact.overflowAmount,
  driftMultiplier: result.boundaryImpact.driftMultiplier,
  phase: result.boundaryImpact.after.phase
});

File: character_os/physics_demo.py
"""Minimal Character Physics V1 demo."""

from __future__ import annotations

from physics_engine import CharacterPhysicsEngine, CharacterPhysicsState, ExperienceEvent
from personality.space import (
    BASE_PERSONALITY_DIMENSIONS,
    big_five_from_coordinate,
    lin_fan_initial_coordinate,
)


def main() -> None:
    """Run one event through the non-prompt physics engine."""

    coordinate = lin_fan_initial_coordinate()
    state = CharacterPhysicsState(
        coordinate=coordinate,
        personality=big_five_from_coordinate(coordinate),
        learning_rate=0.03,
    )
    event = ExperienceEvent(
        id="event_wangxue_no_reply",
        description="王雪已经三天没有回复林凡的消息，今天深夜突然出现在他家门口。",
        tags=["王雪", "失联", "等待", "亲密关系", "夜晚"],
        intensity=0.82,
        importance=0.86,
        relationship_weight=0.95,
        expectation_gap=0.78,
        personality_sensitivity=0.9,
    )

    result = CharacterPhysicsEngine().process_event(state, event)

    print("Character Physics V1 Demo")
    print("\nPersonality Coordinate System:")
    print(f"dimensions={[dimension.key for dimension in BASE_PERSONALITY_DIMENSIONS]}")
    print(f"lin_fan_initial_coordinate={coordinate.to_dict()}")
    print("\nEvent:")
    print(result.event.description)
    print("\nImpact:")
    print(
        f"score={result.impact_score.value}, "
        f"band={result.impact_score.band}, "
        f"description={result.impact_score.description}"
    )
    print("\nEmotion:")
    print(
        f"primary={result.emotion.primary}, "
        f"valence={result.emotion.valence}, "
        f"arousal={result.emotion.arousal}, "
        f"intensity={result.emotion.intensity}"
    )
    print("\nImpact Particle:")
    print(
        f"category={result.particle.category}, "
        f"coordinate_delta={result.particle.vector.delta.to_dict()}, "
        f"rationale={result.particle.vector.rationale}"
    )
    print("\nMemory Node:")
    print(
        f"id={result.memory_node.id}, "
        f"importance={result.memory_node.importance}, "
        f"emotion={result.memory_node.emotion}, "
        f"repetition={result.memory_node.repetition_count}, "
        f"cluster_id={result.memory_node.cluster_id}, "
        f"belief_effect={result.memory_node.belief_effect}"
    )
    print("\nImpact Cluster:")
    print(
        f"id={result.cluster.id}, mass={result.cluster.mass}, "
        f"density={result.cluster.density}, stability={result.cluster.stability}"
    )
    print(f"center_coordinate={result.cluster.center_coordinate.to_dict()}")
    print("\nPersonality Drift:")
    print(f"before={result.drift.before.to_dict()}")
    print(f"force={result.drift.total_force.to_dict()}")
    print(f"after={result.drift.after.to_dict()}")
    print("\nCoordinate Drift:")
    print(f"before={result.coordinate_drift.before.to_dict()}")
    print(f"force={result.coordinate_drift.total_force.to_dict()}")
    print(f"after={result.coordinate_drift.after.to_dict()}")


if __name__ == "__main__":
    main()
